server/src/main/java/org/eclipse/lsp/cobol/service/CobolTextDocumentService.java

```diff
@@ -267,6 +267,16 @@ public void onRunAnalysisEventCallback(@NonNull RunAnalysisEvent event) {
   public CompletableFuture<ExtendedApiResult> analysis(@NonNull JsonObject json) {
     AnalysisResultEvent event = new Gson().fromJson(json.toString(), AnalysisResultEvent.class);
     String uri = Optional.ofNullable(event).map(AnalysisResultEvent::getUri).orElse("");
+    boolean triggerAnalyze = false;
+    synchronized (cfAstMap) {
+      if (!cfAstMap.containsKey(uri) || cfAstMap.get(uri).isDone()) {
```

Comment:
Should it also interrupt the previous analysis?

Reply:
good point

Reply:
It's synchronized. But it's not the concern. Not clear what will happen when a user does request twise. We can't cancel previous computable future because it can be caused by didOpen, and we don't want hanging requests.